src/consensus/upgrades.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2018 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "consensus/upgrades.h"
+
+const struct NUInfo NetworkUpgradeInfo[Consensus::MAX_NETWORK_UPGRADES] = {
+    {
+        /*.nBranchId =*/ 0x00000000,
+        /*.strName =*/ "Test dummy",
+        /*.strInfo =*/ "Test dummy info",
+    }
+};
+
+UpgradeState NetworkUpgradeState(
+    int nHeight,
+    const Consensus::Params& params,
+    Consensus::UpgradeIndex idx)
+{
+    assert(nHeight >= 0);
+    auto nActivationHeight = params.vUpgrades[idx].nActivationHeight;
+
+    if (nActivationHeight == Consensus::NetworkUpgrade::NO_ACTIVATION_HEIGHT) {
+        return UPGRADE_DISABLED;
+    } else if (nHeight >= nActivationHeight) {
+        // From ZIP ???:
```

Comment:
Is there a ZIP for this? This comment needs to be updated.

Reply:
Yes, I will update this comment once @daira assigns a ZIP number to the corresponding draft.

Reply:
It's possible that the caller passes in -1 for nHeight.

Reply:
No it isn't, because of the assertion at the start of the function.